plugins/aws/sagemaker/notebookCmkEncrypted.js

```diff
@@ -0,0 +1,120 @@
+    severity: 'High',
+    description: 'Ensure Notebook instances are encrypted with desired encryption level.',
+    more_info: 'When you encrypt AWS Sagemaker notebook instance using your own AWS Customer Managed Keys (CMKs) for enhanced protection, you have full control over who can use the encryption keys to access your notebook data.',
+    recommended_action: 'Encrypt Notebook Instance with desired encryption level.',
+    link: 'https://docs.aws.amazon.com/sagemaker/latest/dg/encryption-at-rest.html',
+    apis: ['SageMaker:listNotebookInstances','SageMaker:describeNotebookInstance', 'KMS:listKeys', 'KMS:describeKey'],
+    settings: {
+        notebook_desired_encryption_level: {
+            name: 'Notebook Instance Encryption Level',
+            description: 'In order (lowest to highest) awskms=AWS-managed KMS; awscmk=Customer managed KMS; externalcmk=Customer managed externally sourced KMS; cloudhsm=Customer managed CloudHSM sourced KMS',
+            regex: '^(sse|awskms|awscmk|externalcmk|cloudhsm)$',
+            default: 'awscmk',
+        }
+    },
+    realtime_triggers: ['sagemaker:CreateNotebookInstance', 'sagemaker:DeleteNotebookInstance'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+        var config = {
+            desiredEncryptionLevelString: settings.notebook_desired_encryption_level || this.settings.notebook_desired_encryption_level.default
+        };
+
+        var desiredEncryptionLevel = helpers.ENCRYPTION_LEVELS.indexOf(config.desiredEncryptionLevelString);
+        var currentEncryptionLevel;
+
+
+        async.each(regions.sagemaker, function(region, rcb){
+            var listNotebookInstances = helpers.addSource(cache, source,
+                ['sagemaker', 'listNotebookInstances', region]);
+
+            if (!listNotebookInstances) return rcb();
+
+            if (listNotebookInstances.err) {
+                helpers.addResult(results, 3,
+                    'Unable to query for Notebook Instances: ' +
+                    helpers.addError(listNotebookInstances), region);
+                return rcb();
+            }
+
+            if (!listNotebookInstances.data || !listNotebookInstances.data.length) {
+                helpers.addResult(
+                    results, 0, 'No Notebook Instances Found', region);
+                return rcb();
+            }
+
+            var listKeys = helpers.addSource(cache, source,
+                ['kms', 'listKeys', region]);
+
+            if (!listKeys || listKeys.err || !listKeys.data) {
+                helpers.addResult(results, 3,
+                    `Unable to list KMS keys: ${helpers.addError(listKeys)}`, region);
+                return rcb();
+            }
+
+            for (var i in listNotebookInstances.data) {
+                var instance = listNotebookInstances.data[i];
+                var instanceArn = instance.NotebookInstanceArn;
```

Comment:
add the check for following if ( !instance.NotebookInstanceArn || !instance.NotebookInstanceName) continue;